packages/esm-service-queues-app/src/queue-screen/queue-screen.component.tsx

```diff
@@ -1,15 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { DataTableSkeleton } from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { useActiveTickets } from './useActiveTickets';
 import PatientQueueHeader from '../patient-queue-header/patient-queue-header.component';
 import styles from './queue-screen.scss';
+import { useSelectedQueueLocationUuid } from '../helpers/helpers';
 
 interface QueueScreenProps {}
 
 const QueueScreen: React.FC<QueueScreenProps> = () => {
   const { t } = useTranslation();
-  const { activeTickets, isLoading, error } = useActiveTickets();
+  const { activeTickets, isLoading, error, mutate } = useActiveTickets();
+  const speaker = window.speechSynthesis;
+  const [isSpeaking, setIsSpeaking] = useState(false);
+  const selectedLocation = useSelectedQueueLocationUuid();
+  const locationFilteredTickets = activeTickets;
+
+  const rowData = locationFilteredTickets.map((ticket, index) => ({
+    id: `${index}`,
+    room: ticket.room,
+    ticketNumber: ticket.ticketNumber,
+    status: ticket.status,
+  }));
+
+  function readTicket(queue, speaker) {
+    if ('speechSynthesis' in window) {
+      const message = new SpeechSynthesisUtterance();
+      const utterance =
+        'Ticket Number: ' +
+        queue.ticketNumber.split('-')[0].split('') +
+        ', - ' +
+        queue.ticketNumber.split('-')[1].split('') +
+        ', Please Proceed To Room, ' +
+        queue.room;
+      message.rate = 1;
+      message.pitch = 1;
+      message.text = utterance;
+      return new Promise((resolve) => {
+        message.onend = resolve;
+        speaker.speak(message);
+      });
+    } else {
+      return Promise.resolve();
+    }
+  }
+
+  useEffect(() => {
+    const readableTickets = locationFilteredTickets.filter((item) => item.status == 'calling');
```

Comment:
readableTickets may mean some are not readable. We can adopt a name i.e. ticketsToCallOut or something else

Reply:
resolved